src/Event/ControllerListener.php

```diff
@@ -60,13 +60,6 @@ public function onKernelController(FilterControllerEvent $event): void
         $route = $request->attributes->get('_route');
         $controller = $event->getController();
         $service = $this->groupService;
-        $syliusConfig = $request->attributes->get('_sylius');
-        if (!empty($syliusConfig['permission'])) {
-            $syliusPermission = $syliusConfig['permission'];
-            if (!empty($syliusPermission && $syliusPermission === true)) {
-                return;
-            }
-        }
```

Comment:
In my opinion, the only change should be :

if (!empty($syliusPermission) || $syliusPermission === false)) {

Because we don't want routes without permission: true to be validated by this.

Reply:
there is a problem with the hole condition because as you can see in any route
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/AdminBundle/Resources/config/routing
the `permission: true` in all of the routes so you are passing this request another solution is to move this block under
```php
if ($controller[0] instanceof Controller) {
...
}
```

Reply:
Yeah you are right. Let's move the condition under the block then.

Because the change as it is right now will have the side effect of enabling the security on all pages, even the one with no `permission` or `permission: false`